client/src/AddFriend.js

```diff
@@ -33,28 +35,48 @@ class AddFriend extends React.Component {
 
   handleAdd (e) {
     // check for empty inputs
-    // Add new user to users table
-    // Add friend relationship to friendships table
-
-    if (!this.state.firstname || !this.state.lastname || !this.state.email) {
+    this.setState({isloading: true})
+    if (!this.state.friendfname || !this.state.friendlname || !this.state.friendemail) {
       return window.alert('Input field empty')
     }
     axios.post('/api/friendships', this.state)
       .then(res => {
- 
+        if (res.status >= 200) {
+          window.alert('You are now friends with ' + this.state.friendfname + ' ' + this.state.friendlname)
+          this.setState(state => {
+            const friendships = state.friendships.concat(state.friendfname + ' ' + state.friendlname)
```

Comment:
I thought `.concat` is used to merge two lists, but the argument is a string? You're probably looking for `.push`, which appends the string to the end of the list.

Reply:
Apparently `.concat` works for string too. Javascript is weird man.
 https://www.w3schools.com/jsref/jsref_concat_string.asp